File: 0049-group-anagrams/0049-group-anagrams.py
class Solution(object):
    def groupAnagrams(self, strs):
        """
        :type strs: List[str]
        :rtype: List[List[str]]
        """
        str_dict = {}
        for word in strs : 
            w_key = ''.join(sorted(word))
            if w_key not in str_dict :
                str_dict[w_key] = []
            str_dict[w_key].append(word)
        return sorted(str_dict.values(), key=lambda x : len(x))

File: 0049-group-anagrams/0049-group-anagrams.cpp
class Solution {
public:
    vector<vector<string>> groupAnagrams(vector<string>& strs) {
        unordered_map<string, vector<string>> str_dict;
        for (auto s : strs) {
            string tmp = s;
            sort(tmp.begin(), tmp.end());
            str_dict[tmp].push_back(s);
        }
        vector<vector<string>> ans;
        for (auto elem : str_dict) {
            ans.push_back(elem.second);
        }
        return ans;
    }
};